include/nlohmann/detail/meta/cpp_future.hpp

```diff
@@ -152,15 +153,19 @@ template<> struct priority_tag<0> {};
 template<typename T>
 struct static_const
 {
-    static constexpr T value{};
+    static JSON_INLINE_VARIABLE constexpr T value{};
 };
 
 #ifndef JSON_HAS_CPP_17
-
     template<typename T>
-    constexpr T static_const<T>::value; // NOLINT(readability-redundant-declaration)
-
+    constexpr T static_const<T>::value;
 #endif
 
+template<typename T, typename... Args>
+inline constexpr std::array<T, sizeof...(Args)> make_array(Args&& ... args)
```

Comment:
Should this be more like `to_array` from C++20?

Reply:
Maybe. This was initially based on Library TS but isn't needed anymore in newer C++ versions. Let me look at `to_array`...

Reply:
No, it's definitely `make_array` from the Library Fundamentals TS. I put it in `cpp_future.hpp` because C++17 solved this problem by introducing deduction guides.

Reply:
Yes? That's what I said?

> based on Library [Fundamentals] TS but isn't needed anymore in newer C++ versions

> I put it in cpp_future.hpp because C++17 solved this problem by introducing deduction guides.

Reply:
I was referring to the part about `std::to_array` being in C++20.

Reply:
If you want to keep it based on the old `make_array` that's fine too, just pointing out the current standardized name.

Reply:
Okay, I was looking at being the same as this example `std::to_array({ 0, 2, 1, 3 });`